guides/operator/orca-redis-to-sql.md

```diff
@@ -0,0 +1,195 @@
+
+An example of wiring up MariaDB into Orca can be found here: [robzienert/orca-mariadb-extension](https://github.com/robzienert/orca-mariadb-extension).
+
+## Configuring Orca for SQL
+
+This configuration is your baseline for Orca to talk to SQL, in orca.yml:
+
+```yaml
+sql:
+  enabled: true
+  connectionPool:
+    jdbcUrl: jdbc:mysql://localhost:3306/orca
+    user: orca_service
+    password: hunter2
+    connectionTimeout: 5000
+    maxLifetime: 30000
+    # MariaDB-specific:
+    maxPoolSize: 50
+  migration:
+    jdbcUrl: jdbc:mysql://localhost:3306/orca
+    user: orca_migrate
+    password: hunter2
+
+# Ensure we're only using SQL for accessing execution state
+executionRepository:
+  sql:
+    enabled: true
+  redis:
+    enabled: false
+
+# Reporting on active execution metrics will be handled by SQL
+monitor:
+  activeExecutions:
+    redis: false
+```
+
+## Migrating from SQL to Redis
+
+Migrating without downtime from Redis to SQL is a three step process:
+
+1. Deploy Orca with the `DualExecutionRepository` writing to both Redis and SQL.
+2. Deploy a new Orca cluster with migrators enabled and queue processing disabled.
+3. Once all executions have been migrated, delete migration cluster and disable `DualExecutionRepository`.
+
+When `DualExecutionRepository` is running, writes will be routed to either Redis or SQL. Executions will only be migrated to SQL once they've completed (either successfully or terminally): This keeps the migration story simple. As such, the migration agents will need to run for awhile. At Netflix, we ran the migration cluster for two weeks, as we had long pipeline executions due to canaries. You may only need to run the migration cluster for an hour.
+
+### Enabling DualExecutionRepository
+
+Building atop the baseline configuration above, add the following to orca.yml:
+
+```yaml
+executionRepository:
+  dual:
+    enabled: true
+    primaryClass: com.netflix.spinnaker.orca.sql.telemetry.SqlInstrumentedExecutionRepository
+    previousClass: com.netflix.spinnaker.orca.telemetry.RedisInstrumentedExecutionRepository
+  sql:
+    enabled: true
+  redis:
+    enabled: true
+```
+
+Note that all three execution repositories are enabled. Orca will fail to start up if the `DualExecutionRepository` is misconfigured.
+
+### Deploying a Migration Cluster
+
+At Netflix, we deployed `orca-main`, the cluster that serves our production traffic, as well as `orca-main-sqlmigration`, which does not receive API traffic nor processes the work queue. It's sole purpose is to shovel bits from Redis to SQL.
+
+Building atop the configuration for `DualExecutionRepository`, in orca.yml:
+
+```yaml
+pollers:
+  orchestrationMigrator:
+    enabled: true
+    intervalMs: 1800000
+  pipelineMigrator:
+    enabled: true
+    intervalMs: 1800000
+
+queue:
+  redis:
+    enabled: false
+
+keiko:
+  queue:
+    enabled: false
+```
```

Comment:
Is this yaml still the Netflix example, or is this a generic how-to? If still Netflix specific, can you change the heading of this section to 

### Deploying a migration cluster&mdash;a Netflix example

Again, I like using Netflix as an example throughout, but if we make Netflix third person instead of third person we can make it clearer  what parts are generic instruction and what parts are examples of how Netflix does it.

Reply:
This is a generic configuration for use with a stand-alone migration cluster.